src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs

```diff
@@ -56,7 +57,16 @@ public override async Task ExecuteResultAsync(ActionContext context)
         {
             EnsureArg.IsNotNull(context, nameof(context));
 
-            var fhirContext = context.HttpContext.RequestServices.GetService<RequestContextAccessor<IFhirRequestContext>>();
+            RequestContextAccessor<IFhirRequestContext> fhirContext = null;
+
+            try
+            {
+                fhirContext = context.HttpContext.RequestServices.GetService<RequestContextAccessor<IFhirRequestContext>>();
+            }
+            catch (ObjectDisposedException ode)
+            {
+                throw new InstanceUnavailableException("Not able to create final result.", ode);
```

Comment:
Since this message will be seen by the customer should we add details on what action they should take?

Reply:
Seeing the other place this exception is used an action recommendation may want to be added to the exception class so that the it is uniform across usages.

Reply:
I added more information to the exception.
Now the message says: "Not able to create final result. Retry the operation."

Reply:
I replaced InstanceUnavailableException with a ServiceException. That's used in other places. Given that, we'll not be able to standardize the error message.